packages/editor/src/components/block-icon/style.scss

```diff
@@ -9,8 +9,17 @@
 		}
 	}
 
-	svg:not(.dashicon) {
+	// By default, library icons should be 24px by 24px.
+	// If an icon is provided that has a `dashicons` class, or an explicit width/height,
+	// we don't override the height/width.
+	svg:not(.dashicon):not([width]) {
```

Comment:
Based on the comments, shouldn't this be:

```scss
svg:not(.dashicon),
svg:not([height]),
svg:not([width]) {
```

As it is an explicit height alone would still be overridden, and the docs imply that an explicit width OR height will prevent it–even if the `dashicon` class is there. That's how I read it.

Reply:
I think the 24px width/height needs to be applied to the container, otherwise the height of the whole item—when using a smaller icon—is wrong in the inserter:

![edit_page_ _mindful_ _wordpress](https://user-images.githubusercontent.com/1231306/44533986-ba0cc500-a6c4-11e8-95eb-ad3e09255eba.png)

Reply:
> As it is an explicit height alone would still be overridden, and the docs imply that an explicit width OR height will prevent it–even if the dashicon class is there. That's how I read it.

Your point that this comment needs clarification is solid. I got a headache just trying to walk through this now. I might need your help to clarify my code :D 

So, this is what we want to accomplish:

- If you just dump in an SVG and don't really think about it, it's 24x24 as it should be
- If you dump in a dashicon, we know that's 20x20 so we don't enforce 24x24
- If you specify an explicit width (or height, do we have to check for that, seems like we could assume with a width?) — then sure, go ahead and have a custom sized icon so long as it's smalller than 24x24.

Hmm. Actually, your code should do the same shouldn't it?

How would you write this as a comment? Thanks so much for your help!